tests/toolchains/defs.bzl

```diff
@@ -12,192 +12,40 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""This module contains the definition for the toolchains testing rules.
-"""
+""
 
 load("//python:versions.bzl", "PLATFORMS", "TOOL_VERSIONS")
-load("//python/private:bzlmod_enabled.bzl", "BZLMOD_ENABLED")  # buildifier: disable=bzl-visibility
-load("//python/private:toolchain_types.bzl", "TARGET_TOOLCHAIN_TYPE")  # buildifier: disable=bzl-visibility
+load("//tests/support:sh_py_run_test.bzl", "py_reconfig_test")
 
-_WINDOWS_RUNNER_TEMPLATE = """\
-@ECHO OFF
-set PATHEXT=.COM;.EXE;.BAT
-powershell.exe -c "& ./{interpreter_path} {run_acceptance_test_py}"
-"""
+def define_toolchain_tests(name):
+    """Define the toolchain tests.
 
-def _acceptance_test_impl(ctx):
-    files = []
-
-    if BZLMOD_ENABLED:
-        module_bazel = ctx.actions.declare_file("/".join([ctx.attr.python_version, "MODULE.bazel"]))
-        ctx.actions.expand_template(
-            template = ctx.file._module_bazel_tmpl,
-            output = module_bazel,
-            substitutions = {"%python_version%": ctx.attr.python_version},
-        )
-        files.append(module_bazel)
-
-        workspace = ctx.actions.declare_file("/".join([ctx.attr.python_version, "WORKSPACE"]))
-        ctx.actions.write(workspace, "")
-        files.append(workspace)
-    else:
-        workspace = ctx.actions.declare_file("/".join([ctx.attr.python_version, "WORKSPACE"]))
-        ctx.actions.expand_template(
-            template = ctx.file._workspace_tmpl,
-            output = workspace,
-            substitutions = {"%python_version%": ctx.attr.python_version},
+    Args:
+        name: Only present to satisfy tooling.
+    """
+    for platform_key, platform_info in PLATFORMS.items():
+        native.config_setting(
+            name = "_is_{}".format(platform_key),
+            constraint_values = platform_info.compatible_with,
```

Comment:
nit: Platforms also have `flag_values` (e.g. [here](https://github.com/bazelbuild/rules_python/blob/main/python/versions.bzl#L589)). And it would be good to just add them here as well.

For now it does not change how the tests perform, but once we have `musl` toolchains, it may affect things. It is also probably fine to add it then.

Reply:
Created #2106